# Mortalidad por diabetes en Mexico

## Introducción

La diabetes mellitus tipo 2 representa uno de los principales problemas de salud pública en México, tanto por su alta prevalencia como por su impacto en la mortalidad. En este contexto, el análisis sistemático de los registros de defunciones permite comprender la evolución del fenómeno, identificar patrones territoriales y temporales, y apoyar la toma de decisiones en materia de prevención y atención médica.

El Instituto Nacional de Estadística y Geografía (INEGI), en coordinación con la Secretaría de Salud (SSA), pone a disposición bases de datos que registran las causas de muerte en el país, clasificadas conforme a estándares internacionales como la CIE-10. Estos datos constituyen una fuente confiable y estructurada para realizar análisis exploratorios que permitan evaluar la magnitud de las muertes asociadas a la diabetes tipo 2 y su distribución por entidad federativa y año.

Este proyecto tiene como objetivo realizar un Análisis Exploratorio de Datos (EDA) sobre las defunciones por diabetes mellitus tipo 2 en México, utilizando información publicada por el INEGI. La exploración se centrará en:

 + Identificar tendencias temporales en la mortalidad.

 + Comparar la incidencia entre entidades federativas.

 + Detectar variaciones relevantes y posibles anomalías en los registros.

 + Generar visualizaciones que faciliten la interpretación de los datos.

El análisis busca ofrecer una visión clara y fundamentada del comportamiento de esta causa de muerte en el país, proporcionando insumos útiles para investigaciones posteriores, diseño de políticas públicas y estrategias de intervención en salud.

## Objetivo general

Analizar de manera exploratoria los registros de defunciones por diabetes mellitus tipo 2 en México, utilizando datos publicados por el INEGI y generados por la Secretaría de Salud (SSA), con el fin de identificar patrones temporales y espaciales que contribuyan a la comprensión del impacto de esta enfermedad en las entidades federativas del país.

## Objetivos específicos

 + Examinar la distribución anual de muertes por diabetes tipo 2 para identificar tendencias y variaciones relevantes.

 + Comparar la mortalidad entre entidades federativas para detectar diferencias territoriales significativas.

 + Evaluar la calidad, completitud y consistencia del dataset mediante técnicas de análisis exploratorio.

 + Generar visualizaciones que faciliten la interpretación de los patrones observados.

 + Proponer hallazgos preliminares que puedan servir como base para estudios posteriores o para la formulación de políticas públicas en salud.

## Justificación del proyecto

La diabetes mellitus tipo 2 es una de las principales causas de muerte en México y representa un desafío persistente para el sistema de salud. Comprender cómo se comporta la mortalidad asociada a esta enfermedad a lo largo del tiempo y entre las distintas entidades federativas es fundamental para diseñar estrategias de prevención, atención y asignación de recursos.

Los datos de defunciones publicados por el INEGI, basados en los certificados médicos de la SSA, constituyen una fuente confiable y estandarizada que permite realizar análisis estadísticos rigurosos. Un Análisis Exploratorio de Datos (EDA) es el primer paso indispensable para revelar patrones, anomalías y relaciones que no son evidentes a simple vista. Este proyecto aporta valor al transformar datos crudos en información útil para la investigación, la docencia y la toma de decisiones en salud pública.

## Metodología del EDA

El análisis exploratorio se desarrollará siguiendo una secuencia lógica de etapas:

1. Obtención y carga del dataset

    + Descarga del archivo desde la plataforma del INEGI.

    + Importación del dataset en un entorno de análisis (Python, R o similar).

2. Revisión inicial de la estructura de los datos

    + Identificación de columnas, tipos de datos y metadatos relevantes.

    + Verificación de la presencia de claves como entidad federativa, año y causa de muerte (CIE-10).

3. Limpieza y preparación del dataset

    + Manejo de valores faltantes.

    + Corrección de inconsistencias en nombres de estados o códigos.

    + Filtrado de registros correspondientes a diabetes mellitus tipo 2 (CIE-10: E11).

4. Análisis descriptivo

    + Cálculo de frecuencias por año y por entidad.

    + Identificación de máximos, mínimos y variabilidad.

    + Detección de outliers o cambios abruptos.

5. Visualización de datos

    + Gráficas de líneas para tendencias temporales.

    + Mapas coropléticos para distribución geográfica.

    + Barras comparativas entre entidades federativas.

6. Interpretación de hallazgos

    + Síntesis de patrones relevantes.

    + Discusión de posibles factores asociados.

    + Señalamiento de áreas que requieren análisis más profundo.


## Obtención de los datos

La base de datos utilizada en este proyecto proviene del sistema de descargas del Instituto Nacional de Estadística y Geografía (INEGI), el cual publica anualmente los registros de defunciones generados por la Secretaría de Salud (SSA). Estos datos se encuentran clasificados conforme a la Clasificación Internacional de Enfermedades (CIE-10), lo que permite identificar de manera precisa las muertes asociadas a diabetes mellitus tipo 2 (código E11).

Para este análisis, se accedió a la plataforma de descargas del INEGI, donde se seleccionó el conjunto de datos correspondiente a las estadísticas de mortalidad. El archivo se obtuvo en formato CSV, lo que facilita su procesamiento en herramientas de análisis como Python o R. Una vez descargado, el dataset fue almacenado localmente y posteriormente cargado en el entorno de trabajo para su revisión, limpieza y análisis.

El proceso de obtención incluyó:

    + Identificación del recurso adecuado dentro del catálogo de descargas del INEGI.

    + Verificación del formato disponible y selección del archivo CSV.

    + Descarga del archivo y comprobación de su integridad.

    + Revisión inicial de los metadatos proporcionados por el INEGI para comprender la estructura del dataset.

Este procedimiento garantiza que los datos utilizados provienen de una fuente oficial, confiable y actualizada, lo que permite realizar un análisis exploratorio riguroso y reproducible.

link: "https://www.inegi.org.mx/app/descarga/ficha.html?tit=3161201&ag=0&f=csv"

## Revision de dataset

In [1]:
import pandas as pd

df_raw = pd.read_csv("../data/raw/Defunciones.csv", encoding="latin-1")

print(df_raw.info())

C:\Users\mario\AppData\Local\Temp\ipykernel_8956\2752304382.py:3: DtypeWarning: Columns (68) have mixed types. Specify dtype option on import or set low_memory=False.
  df_raw = pd.read_csv("../data/raw/Defunciones.csv", encoding="latin-1")


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 819672 entries, 0 to 819671
Data columns (total 74 columns):
 #   Column      Non-Null Count   Dtype  
---  ------      --------------   -----  
 0   ent_regis   819672 non-null  int64  
 1   mun_regis   819672 non-null  int64  
 2   tloc_regis  819672 non-null  int64  
 3   loc_regis   819672 non-null  int64  
 4   ent_resid   819672 non-null  int64  
 5   mun_resid   819672 non-null  int64  
 6   tloc_resid  819672 non-null  int64  
 7   loc_resid   819672 non-null  int64  
 8   ent_ocurr   819672 non-null  int64  
 9   mun_ocurr   819672 non-null  int64  
 10  tloc_ocurr  819672 non-null  int64  
 11  loc_ocurr   819672 non-null  int64  
 12  causa_def   819672 non-null  object 
 13  cod_adicio  819672 non-null  object 
 14  lista_mex   819672 non-null  object 
 15  sexo        819672 non-null  int64  
 16  ent_nac     819672 non-null  int64  
 17  afromex     819672 non-null  int64  
 18  conindig    819672 non-null  int64  
 19  le

## Interpretación del dataset crudo antes de la limpieza

El dataset original contiene 819,672 registros y 74 columnas, lo que lo convierte en un conjunto de datos amplio, complejo y altamente detallado. Su estructura refleja la naturaleza administrativa y epidemiológica de los registros de defunciones en México, integrando información sobre:

 + Lugar de registro, residencia y ocurrencia (entidad, municipio, localidad).

 + Características demográficas (sexo, edad, lengua indígena, afrodescendencia).

 + Causas de muerte (CIE‑10 principal y códigos adicionales).

 + Variables complementarias (condición indígena, tipo de localidad, indicadores regionales).



+ Completitud.

Todas las columnas muestran 819,672 valores no nulos, lo que indica que el dataset está completamente lleno y no presenta vacíos estructurales. Esto es típico en bases administrativas del INEGI, donde cada registro debe cumplir con un formato estándar.

+ Tipos de datos

    - 68 columnas numéricas (int64)

    - 5 columnas categóricas (object)

    - 1 columna float64

La predominancia de variables numéricas sugiere que el dataset está diseñado para análisis estadísticos y geográficos detallados, aunque también implica que muchas variables requieren decodificación para ser interpretadas correctamente.

+ Complejidad del dataset

El dataset incluye múltiples niveles de detalle territorial:

            Entidad → Municipio → Localidad → Tipo de localidad → Código de localidad

Esto permite análisis espaciales muy finos, pero también introduce redundancia si el objetivo del proyecto es trabajar únicamente a nivel estatal.

Además, contiene variables que no son necesarias para un análisis de diabetes tipo 2, como:

    - Condición afrodescendiente

    - Condición indígena

    - Lengua

    - Indicadores regionales específicos (ej. dis_re_oax)

Estas variables pueden ser útiles para estudios sociodemográficos, pero no son esenciales para tu objetivo actual.

+ Causa de muerte

La columna causa_def está completa y en formato texto, lo que facilita el filtrado por el código CIE‑10 correspondiente a diabetes mellitus tipo 2 (E11).
Esto confirma que el dataset es adecuado para extraer únicamente las defunciones relevantes.

## Síntesis final

El dataset crudo es demasiado grande y complejo para trabajar directamente, pero perfectamente adecuado como fuente para construir un dataset limpio, enfocado y analíticamente útil.
Después de la limpieza y filtrado, el proyecto puede desarrollarse sin limitaciones técnicas.